lib/elixir/lib/code/identifier.ex

```diff
@@ -214,8 +214,11 @@ defmodule Code.Identifier do
   @doc """
   Escapes the given identifier.
   """
-  def escape(other, char, count \\ :infinity, fun \\ &escape_map/1) do
-    escape(other, char, count, [], fun)
+  @spec escape(binary(), char(), :infinity | non_neg_integer, (char() -> iolist() | false)) ::
```

Comment:
```suggestion
  @spec escape(binary(), char(), :infinity | non_neg_integer, (char() -> char() | false)) ::
```

Reply:
Look José.
https://github.com/elixir-lang/elixir/blob/846147285b614f66677ce38d89d00e5038774d53/lib/elixir/lib/code/identifier.ex#L276-L286

Reply:
could be `charlist()` instead, the thing is that chars are not part of io_lists, so it should be [byte()].. That is why iolist made more sense  to me.

Reply:
But it can return chars too, so i think it is `char() | iolist() | false`. WDYT?

Reply:
And good catch btw!

Reply:
Off the top of my head, I think if you return a non-byte char, I think IO.iodata_to_binary() will crash. I think that is why I decided to spec the functions because i could not fully understand how it worked.
Or maybe we should modify the code to deal with chars too?

Reply:
No, you are correct. My bad. :)